tests/api2/test_190_filesystem.py

```diff
@@ -97,3 +100,18 @@ def test_04_test_filesystem_statfs_fstype(pool):
     cmd4 = f'rmdir {nested_path}'
     results = SSH_TEST(cmd4, user, password, ip)
     assert results['result'] is True, results['output']
+
+
+@pytest.mark.parametrize('pool', pool_name)
+def test_05_set_immutable_flag_on_path(pool):
+    def ensure_result_is_sane(results, set):
+        assert results.status_code == 200, results.text
+        result = results.json()
+        assert isinstance(result, bool) is True, results.text
+        assert result == set, 'Immutable flag is still not set' if set else 'Immutable flag is still set'
+
+    t_path = os.path.join('/mnt', pool, 'random_directory_immutable')
+    with directory(t_path) as d:
+        for v in (True, False):
+            POST('/filesystem/set_immutable/', {'set_flag': v, 'path': d})
+            ensure_result_is_sane(POST('/filesystem/is_immutable_set/', d), v)
```

Comment:
What if this flag implementation is broken and just does nothing? Won't it be easier and more reliable to check if the flag is doing the job by just trying to perform an actual write?